src/io/metadata.cpp

```diff
@@ -396,6 +481,23 @@ void Metadata::SetQuery(const data_size_t* query, data_size_t len) {
   #endif  // USE_CUDA_EXP
 }
 
+void Metadata::InsertQueries(const data_size_t* queries, data_size_t start_index, data_size_t len) {
```

Comment:
may be we can extract the common codes in InsertLabels/InsertWeights/InsertQueries, and reuse them, to avoid the duplicated codes.

Reply:
There are different datatypes and different error messages. I suppose we can, with some templating, but IMHO it's not worth it and doesn't really simplify the code or make it more readable.  If you really want me to I can.  The corresponding SetX() routines are not sharing the same code despite some similarities.